docs/guides/testing/testing.rst

```diff
@@ -117,143 +117,64 @@ Note -
+
+    pytest test --rootdir <lms or cms>
+
+Various tools like ddt create tests with very complex names, rather than figuring out the name yourself, you can:
 
-To select tests to run based on their name, provide an expression to the
-`pytest -k option`_ which performs a substring match on test names::
+1. Select tests to run based on their name, provide an expression to the `pytest -k option`_ which performs a substring match on test names::
 
     pytest common/lib/xmodule/xmodule/tests/test_stringify.py -k test_stringify
 
 .. _pytest -k option: https://docs.pytest.org/en/latest/example/markers.html#using-k-expr-to-select-tests-based-on-their-name
 .. _node ID: https://docs.pytest.org/en/latest/example/markers.html#node-id
 
-Alternatively, you can select tests based on their `node ID`_ directly,
-which is useful when you need to run only one of mutliple tests with the same
-name in different classes or files.
 
-This command runs any python unit test method that matches the substring
-`test_stringify` within a specified TestCase class within a specified file::
+2. Alternatively, you can the get the name of all test methods in a class, file, or project, including all ddt.data variations, by running pytest with `--collectonly`::
 
-    pytest common/lib/xmodule/xmodule/tests/test_stringify.py::TestCase -k test_stringify
-
-Note: if the method has an `@ddt.data` decorator, ddt will create multiple
-methods with the same prefix name and each individual data input as the suffix
-(e.g. `test_stringify_1_foo`). To test all of the ddt.data variations of the
-same test method, pass the prefix name to the pytest `-k` option.
+    pytest common/lib/xmodule/xmodule/tests/test_stringify.py --collectonly
 
-If you need to run only one of the test variations, you can the get the
-name of all test methods in a class, file, or project, including all ddt.data
-variations, by running pytest with `--collectonly`::
 
-    pytest common/lib/xmodule/xmodule/tests/test_stringify.py --collectonly
+.. _the pdb documentation: http://docs.python.org/library/pdb.html
 
+Very handy: if you pass the ``--pdb`` flag to a pytest, the test runner will drop you
+into pdb on error. This lets you go up and down the stack and see what the
+values of the variables are. Check out `the pdb documentation`_.  Note that
+this only works if you aren't collecting coverage statistics (pdb and
+coverage.py use the same mechanism to trace code execution).
 
-This is an example of how to run a single test and get stdout shown immediately, with proper env config::
+Use this command to put a temporary debugging breakpoint in a test.
+If you check this in, your tests will hang on jenkins::
 
-    pytest cms/djangoapps/contentstore/tests/test_import.py -s
+    breakpoint()
```

Comment:
I would also include an example of how to use this. Does it also drop you into pdb? Do you need to import anything to get it?

Reply:
Yep, it drops you into pdb by default and no imports are required